test/integration/testStreamRows.js

```diff
@@ -462,3 +464,91 @@ describe('Test Stream Rows HighWaterMark', function ()
     });
   });
 });
+
+describe('Test stream row modes', function () {
+  this.timeout(18000);
+  let connection;
+  const sql = 'select * ' +
+    'from (select \'a\' as key, 1 as foo, 3 as name) as table1 ' +
+    'join (select \'a\' as key, 2 as foo, 3 as name2) as table2 ' +
+    'on table1.key = table2.key'
+  const expectedArray = ['a', 1, 3, 'a', 2, 3]
+  const expectedObject = {KEY_0: 'a', FOO_1: 1, NAME: 3, KEY_3: 'a', FOO_4: 2, NAME2: 3}
+
+  const createItCallback = function (testCase) {
```

Comment:
maybe let's explicitly define parameters instead of test case Object? it can be done by destructuring also